pysgrid/read_netcdf.py

```diff
@@ -95,11 +95,88 @@ def find_grid_topology_vars(self):
+                if nc_var_location == location_str:
+                    search_results.append(nc_var)
+            except AttributeError:
+                continue
+        return search_results
+    
+    def find_coordinates_by_location(self, location_str, topology_dim):
+        """
+        Find a grid coordinates variables with a location attribute equal
+        to location_str. This method can be used to infer edge, face, or
+        volume coordinates from the location attribute of a variable.
+        
+        Location is a required attribute per SGRID conventions.
+        
+        :param str location_str: the location value to search for
+        :param int topology_dim: the topology dimension of the grid
+        
+        """
+        nc_vars = self.ncd.variables
+        vars_with_location = self.search_variables_by_location(location_str)
+        x_coordinate = None
+        y_coordinate = None
+        z_coordinate = None
+        for var_with_location in vars_with_location:
+            location_var = nc_vars[var_with_location]
+            location_var_dims = location_var.dimensions
+            try:
+                location_var_coordinates = location_var.coordinates
+                lvc_split = location_var_coordinates.split(' ')
+                if len(lvc_split) == 2:
+                    x_coordinate, y_coordinate = lvc_split
+                elif len(lvc_split) == 3:
+                    x_coordinate, y_coordinate, z_coordinate = lvc_split
+                else:
+                    raise TopologyDimensionError(topology_dim)
+                break
+            except AttributeError:
+                # run through this if a location attributed is defined, but not coordinates
+                potential_coordinates = []
+                for nc_var in nc_vars.keys():
+                    nc_var_obj = nc_vars[nc_var]
+                    nc_var_dim_set = set(nc_var_obj.dimensions)
+                    if (nc_var_dim_set.issubset(location_var_dims) and 
+                        nc_var != var_with_location and 
+                        len(nc_var_dim_set) > 0
+                        ):
+                        potential_coordinates.append(nc_var_obj)
+                for potential_coordinate in potential_coordinates:
+                    pc_name = potential_coordinate.name
+                    if 'lon' in pc_name.lower():
+                        x_coordinate = pc_name
+                    elif 'lat' in pc_name.lower():
+                        y_coordinate = pc_name
+                    else:
+                        z_coordinate = pc_name  # this might not always work... 
+        if topology_dim == 2:
+            coordinates = (x_coordinate, y_coordinate)
+        else:
+            coordinates = (x_coordinate, y_coordinate, z_coordinate)
+        if all(coordinates):
+            coordinate_result = coordinates
+        else:
+            coordinate_result = None
+        return coordinate_result
+    
+    @deprecated
```

Comment:
This library is in such an earlier that there is not a release version, nor it is been used in production anywhere.  So I don't think you need deprecation.  Lets keep the code base clean and remove it.